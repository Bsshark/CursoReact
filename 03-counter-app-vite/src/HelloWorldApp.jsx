//export function App() {
    // document.createElement...

//    return <h1> Hola Mundo !!</h1>;
//} 

//rafc
export const HelloWorldApp = () => {
  return (
    <h1>HelloWorld App</h1>
  )
}
