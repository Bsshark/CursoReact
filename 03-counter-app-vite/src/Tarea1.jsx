import { Fragment } from "react"



const newMessage = {
  message: 'Hola mundo',
  title: 'Abraham'
}

function getMessage() {
  return newMessage.message;
}

export const FirstApp = () => {

  return (
    <>
    <h2>{ getMessage()  }</h2>
    <p>Ey</p>
    </>
  )
}
