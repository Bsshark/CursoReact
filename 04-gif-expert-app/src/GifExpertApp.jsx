import React, { useState } from 'react';
import { AddCategory } from './Components/AddCategory';

export const GifExpertApp = () => {

  const [categories, setCategories] = useState([ 'One Punch', 'Dragon Ball' ]);

  const onAddCategory = () => {
    setCategories(['Valorant', ...categories]);
  }

  return (
    <>
      {/* TÃ­tulo */ }
        <h1>GifExpertApp</h1>

        { /* Input */ }
        <AddCategory />

        { /* Tarjetas */ }
        <button onClick={onAddCategory}>Agregar</button>
        <ol>
          { categories.map( category => {
            return <li key={ category }>{category}</li>
          })}
        </ol>
          { /* GifItem */ }
    </>
  )
}
