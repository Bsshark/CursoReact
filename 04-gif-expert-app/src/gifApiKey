dZ57QsWCPgRynsb2TvZeW2FvDP8JfeHw