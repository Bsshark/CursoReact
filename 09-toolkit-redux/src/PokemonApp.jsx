import { useEffect } from "react"
import { useDispatch, useSelector, useStore } from "react-redux"
import { getPokemons } from "./store/slices/pokemon";
import { store } from "./store";

export const PokemonApp = () => {

  const dispatch = useDispatch();
  const {pokemons = [], isLoading, page} = useSelector(state => state.pokemons);
  

  useEffect(() => {
    dispatch( getPokemons() );
  }, [])

  

  
  

  return (
    <>
        <h1>Pokemon App</h1>
        <hr />

        <span>Loading: {isLoading ? `True`: `False`}</span>

        <ul>
          {
            pokemons.map((pokemon) => <li key={pokemon.url}>{pokemon.name.charAt(0).toUpperCase() + pokemon.name.slice(1)}</li>)
          }
        </ul>

        {<button disabled={isLoading} onClick={() => dispatch(getPokemons(page))}>
          Next
        </button>}
    </>
  )
}
