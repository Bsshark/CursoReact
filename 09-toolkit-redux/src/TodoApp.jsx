import React, { useState } from 'react'
import { useGetTodoByIdQuery, useGetTodosQuery } from './store/apis'

export const TodoApp = () => {

  /* const {data: todos = [], isLoading, isFetching} = useGetTodosQuery(); */
  

 
  const [todoId, settodoId] = useState(1);
  const { data: todo, isLoading} = useGetTodoByIdQuery(todoId);

  const nextToDo = () => {
    settodoId(todoId + 1);
  }

  const prevToDo = () => {
    if(todoId != 1) {
      settodoId(todoId - 1);
    }
    
  }
  

  return (
    <>
        <h1>TODO - RTK Query</h1>
        <hr/>

        <h4>isLoading: {isLoading ? 'true' : 'false' }</h4>
        <pre>{JSON.stringify(todo)}</pre>



        {/* <ul>
          {todos.map(todo => (
            <li key={todo.id}>
              <strong>{todo.completed ? 'DONE' : 'PENDING'}</strong> <br />
              {todo.title}
              <hr />
            </li>
          ))}
        </ul> */}

        <button onClick={prevToDo}>
            Previous TODO
        </button>
        <button onClick={nextToDo}>
            Next TODO
        </button>
    </>
  )
}
