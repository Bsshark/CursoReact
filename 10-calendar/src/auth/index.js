export * from './index';
export * from './pages/LoginPage';
