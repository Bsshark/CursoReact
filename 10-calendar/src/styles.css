.fab {
  border-radius: 100%;
  bottom: 25px;
  font-size: 20px;
  padding: 20px;
  position: fixed;
  right: 25px;
}
.fab-danger {
  border-radius: 100%;
  bottom: 25px;
  font-size: 20px;
  position: fixed;
  left: 25px;
  width: 70px;
  height: 70px;
}