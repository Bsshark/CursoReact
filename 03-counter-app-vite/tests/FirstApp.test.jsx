import {render} from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en First App', () => { 
    test('Debe de hacer match con el snapshot', () => {
        const title = 'Soy Goku';
        const { container } = render(<FirstApp title={title}/>);

        expect(container).toMatchSnapshot();

     })

     test('debe mostrar el tÃ­tulo en un h1', () => { 
        const title = 'Soy Goku'
        const { container, getByText } = render(<FirstApp title={title}/>);

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toBe(title);
      })
 });