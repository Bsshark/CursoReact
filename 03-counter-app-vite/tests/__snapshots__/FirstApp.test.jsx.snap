// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en First App Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Abraham de Le√≥n
  </p>
</div>
`;
